### PR TITLE
DashboardQuery: Expand query options

### PR DESCRIPTION
**Pull request by ryantxu**
This extends the existing "-- Dashboard --" query, so that it:
1. Can optionally return data *after* transform, not before
2. Can select the "annotations" from another panel and show them directly

image

image

TODO:
- [x] Add devenv... see http://localhost:3000/d/fYGWTVW4k/reuse-dashboard-queries?orgId=1
- [x] Unit test
- [x] Update support snapshots to use this option to show transforms and annotations

### CODE REVIEW

File: public/app/plugins/datasource/dashboard/runSharedRequest.ts [hunk 1]
```diff
@@ -49,11 +50,24 @@ export function runSharedRequest(options: QueryRunnerOptions): Observable<PanelD
...
+          if (query?.topic === DataTopic.Annotations) {
+            data = {
+              ...data,
+              series: data.annotations ?? [],
+              annotations: undefined, // remove annotations
+            };
+          }
+          subscriber.next(data);
+        },
+      });
```
Comment (torkelo) [inline public/app/plugins/datasource/dashboard/runSharedRequest.ts:70]: Would be nice with new unit tests that cover this, but runSharedRequest did not have any to start from. But does not look that tricky.

Comment (ryantxu) [reply public/app/plugins/datasource/dashboard/runSharedRequest.ts:70]: done 🎉

File: public/app/plugins/datasource/dashboard/DashboardQueryEditor.tsx [hunk 2]
```diff
@@ -131,19 +156,45 @@ export function DashboardQueryEditor({ panelData, queries, onChange, onRunQuerie
...
+      ) : (
+        <>
+          {results && Boolean(results.length) && (
+            <Field label="Queries">
+              <VerticalGroup spacing="sm">
+                {results.map((target, i) => (
+                  <div className={styles.queryEditorRowHeader} key={`DashboardQueryRow-${i}`}>
+                    <div>
+                      <img src={target.img} width={16} />
+                      <span className={styles.refId}>{`${target.refId}:`}</span>
```
Comment (torkelo) [inline public/app/plugins/datasource/dashboard/DashboardQueryEditor.tsx:172]:
```suggestion
                      <span className={styles.refId}>{target.refId}:</span>
```

Same no? :) just nitpick

File: public/app/plugins/datasource/dashboard/runSharedRequest.ts (hunk 1 above)
Comment (torkelo) [reply public/app/plugins/datasource/dashboard/runSharedRequest.ts:70]: nice!

**Review by torkelo — APPROVED** [review]

**Pull request merged**